components/applications-service/pkg/storage/postgres/services.go

```diff
@@ -114,7 +114,7 @@ LEFT JOIN deployment AS d
 LEFT JOIN supervisor AS sup
   ON s.sup_id = sup.id
  %s
-ORDER BY %s %s
+ORDER BY %s %s, sup_member_id ASC
```

Comment:
Why do we add this?

Reply:
In case anyone else is wondering:

The primary sort here (or before this patch, the only sort) is on health status, which has a lot of elements that are equal. The order that results are returned in this case is defined by Postgres implementation details (probably whatever is most efficient to read from disk or something like that); I found that adding an insert _to a different table_ was changing the order that results got returned in a few of the integration tests. Adding this secondary sort (on a column that will be unique for each element in this set) makes the ordering of the results completely deterministic.

If the order was changing between requests, then that would be a pagination bug, but I don't think that was happening. In any case, this change means there is one correct order for the results which I think is better than not having a correct ordering if we're already doing a sort anyway.